agent/src/beerocks/slave/tasks/topology_task.cpp

```diff
@@ -83,6 +83,104 @@ void TopologyTask::work()
     }
 }
 
+bool TopologyTask::handle_cmdu(ieee1905_1::CmduMessageRx &cmdu_rx, const sMacAddr &src_mac,
+                               std::shared_ptr<beerocks_header> beerocks_header)
+{
+    switch (cmdu_rx.getMessageType()) {
+    case ieee1905_1::eMessageType::TOPOLOGY_DISCOVERY_MESSAGE: {
+        handle_topology_discovery(cmdu_rx, src_mac);
+        break;
+    }
+    default: {
+        // Message was not handled, therfore return false.
+        return false;
+    }
+    }
+    return true;
+}
+
+void TopologyTask::handle_topology_discovery(ieee1905_1::CmduMessageRx &cmdu_rx,
+                                             const sMacAddr &src_mac)
+{
+    auto tlvAlMac = cmdu_rx.getClass<ieee1905_1::tlvAlMacAddress>();
+    if (!tlvAlMac) {
+        LOG(ERROR) << "getClass tlvAlMacAddress failed";
+        return;
+    }
+
+    auto db = AgentDB::get();
+
+    // Filter out the messages we have sent.
+    if (tlvAlMac->mac() == db->bridge.mac) {
+        return;
+    }
+
+    auto mid = cmdu_rx.getMessageId();
+    LOG(INFO) << "Received TOPOLOGY_DISCOVERY_MESSAGE from AL MAC=" << tlvAlMac->mac()
+              << ", mid=" << std::hex << mid;
+
+    auto tlvMac = cmdu_rx.getClass<ieee1905_1::tlvMacAddress>();
+    if (!tlvMac) {
+        LOG(ERROR) << "getClass tlvMacAddress failed";
+        return;
+    }
+
+    uint32_t if_index                      = message_com::get_uds_header(cmdu_rx)->if_index;
+    std::string local_receiving_iface_name = network_utils::linux_get_iface_name(if_index);
+    if (local_receiving_iface_name.empty()) {
+        LOG(ERROR) << "Failed getting interface name for index: " << if_index;
+        return;
+    }
+
+    std::string local_receiving_iface_mac_str;
+    if (!network_utils::linux_iface_get_mac(local_receiving_iface_name,
+                                            local_receiving_iface_mac_str)) {
+        LOG(ERROR) << "Failed getting MAC address for interface: " << local_receiving_iface_name;
+        return;
+    }
+
+    // Check if it is a new device so if it does, we will send a Topology Notification.
+    bool new_device = false;
+    for (auto &neighbors_on_local_iface_entry : db->neighbor_devices) {
+        auto &neighbors_on_local_iface = neighbors_on_local_iface_entry.second;
+        new_device =
+            neighbors_on_local_iface.find(tlvAlMac->mac()) == neighbors_on_local_iface.end();
+        if (new_device) {
+            break;
+        }
+    }
```

Comment:
I know this code is ported as-is and maybe it is just me - but I think there must be a clearer way to implement this check.
one example would be:
```
if (neighbors_on_local_iface.find(tlvAlMac->mac()) == neighbors_on_local_iface.end()) {
    new_device = true;
    break;
}
```
a better way is a helper function to check "is_device_in_neigbors_map" or similar and call it here

Reply:
What you've suggested is the same thing as what is written now.
Calling another function just for doing basic `find` on a container is redundant IMO.